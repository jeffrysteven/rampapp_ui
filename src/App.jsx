import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Routes } from './routes';
import { Navigation } from './components/Navigation';
import styles from './App.css';

const App = () => (
  <div className={styles.main}>
    <Router>
      <Navigation />
      <section>
        <Routes />
      </section>
    </Router>
  </div>
);

export default App;
