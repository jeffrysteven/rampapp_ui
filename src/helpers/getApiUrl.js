export function getApiUrl(){
    return 'http://54.157.141.205';
}
