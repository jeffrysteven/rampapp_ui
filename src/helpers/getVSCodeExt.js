const MARKETPLACEVSCODE_URL =
  'https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery';

const getRequestOptions = (query) => {
  const headers = new Headers();
  headers.append('X-Market-User-Id', 'redacted-just-in-case');
  headers.append(
    'accept',
    'application/json;api-version=6.0-preview.1;excludeUrls=true',
  );
  headers.append('Origin', 'file://');
  headers.append(
    'User-Agent',
    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Code/1.40.2 Chrome/76.0.3809.146 Electron/6.1.5 Safari/537.36',
  );
  headers.append('X-Market-Client-Id', 'VSCode 1.40.2');
  headers.append('Sec-Fetch-Mode', 'cors');
  headers.append('Content-Type', 'application/json');
  headers.append(
    'Cookie',
    'VstsSession=%7B%22PersistentSessionId%22%3A%228d4d504e-34f4-4dcb-90ce-40eb492d9437%22%2C%22PendingAuthenticationSessionId%22%3A%2200000000-0000-0000-0000-000000000000%22%2C%22CurrentAuthenticationSessionId%22%3A%2200000000-0000-0000-0000-000000000000%22%2C%22SignInState%22%3A%7B%7D%7D; Gallery-Service-UserIdentifier=2bcc109c-cbf2-4505-ad57-c701aea90487',
  );
  const raw = JSON.stringify({
    filters: [
      {
        criteria: [
          { filterType: 8, value: 'Microsoft.VisualStudio.Code' },
          { filterType: 12, value: '4096' },
          { filterType: 10, value: query },
        ],
        pageNumber: 1,
        pageSize: 50,
        sortBy: 0,
        sortOrder: 0,
      },
    ],
    assetTypes: [],
    flags: 914,
  });

  return {
    method: 'POST',
    headers: headers,
    body: raw,
    redirect: 'follow',
  };
};

export const getVSCodeExtensions = async (query) => {
  try {
    const requestOptions = getRequestOptions(query);
    const response = await fetch(MARKETPLACEVSCODE_URL, requestOptions);
    const result = await response.json();

    return result;
  } catch (error) {
    console.error(error);
  }
};
