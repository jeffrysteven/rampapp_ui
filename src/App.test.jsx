import React from 'react';

import { shallow } from 'enzyme';

import './helpers/enzyme_setup';

import App from './App';

describe('App', () => {
  let component;

  beforeEach(() => {
    component = shallow(<App />);
  });

  it('should render', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
