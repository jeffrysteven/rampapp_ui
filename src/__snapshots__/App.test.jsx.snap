// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div
  className="main"
>
  <BrowserRouter>
    <Navigation />
    <section>
      <Routes />
    </section>
  </BrowserRouter>
</div>
`;
